testsuite/non_trivial_obj_def.tcc

```diff
@@ -0,0 +1,94 @@
+/** for page size defs  **/
+#include <unistd.h>
+/** for numeric limits  **/
+#include <limits>
+/** for std::size_t     **/
+#include <cstddef>
+
+/** place all of this in raft **/
+namespace raft
+{
+/** and inside namespace test **/
+namespace test
+{
+
+struct base_non_trivial_object
+{
+    using ptr_t = std::int64_t;
+    using self_t = const base_non_trivial_object;
+
+    base_non_trivial_object() = default;
+    
+    base_non_trivial_object( ptr_t *value )
+    {
+        if( value != nullptr )
+        {
+            /** 
+             * silly bug...fixed, was assigning  
+             * value <- value vs. setting this 
+             * likely introduced when we made this
+             * test case generic, fixed now. was causing
+             * nonTrivialAllocatePop to fail 
+             * -jcb 1 Sept 2019
+             */
+            (this)->value = value;
+        }
+    }
+
+    ~base_non_trivial_object()
+    {
+        /** no cast needed, should return a 'ptr' val **/
+        *(this)->value = std::numeric_limits< ptr_t >::max();
+    }
+
+    constexpr base_non_trivial_object( self_t &other )
+    {
+        /** let's pass the pointer to this object so we can see if it's deleted **/
+        (this)->value = other.value;
+    }
+
+    constexpr base_non_trivial_object& operator= ( self_t &other )
+    {
+        (this)->value = other.value;
+        return( *this );
+    }
+
+    ptr_t *value = nullptr;
+};
+
+
+/**
+ * for test cases, default gets  you a big pad to initiate external
+ * allocation, otherwise specify the size directly. 
+ */
+template < bool PAD > struct non_trivial_object : base_non_trivial_object{
+};
+
+template <> struct non_trivial_object<true> : base_non_trivial_object
+{
+    non_trivial_object() : base_non_trivial_object(){};
+    
+    non_trivial_object( base_non_trivial_object::ptr_t *ptr ) : base_non_trivial_object( ptr ){};
+    
+    char padding[ 
+#ifdef PAGE_SIZE
+    PAGE_SIZE << 1 
```

Comment:
This may cause problems. According to [this ](http://man7.org/linux/man-pages/man3/sysconf.3.html) we need at least to check if the constant is available and if it is not, we need to fall back to sysconf() to get the current page size. Using sysconf may even be the preferred way (see this [answer](https://stackoverflow.com/questions/37897645/page-size-undeclared-c/37897870#37897870)). Also, on Windows there is no compile time method to obtain page sizes. 

I suggest to leave this for now, just include the "unistd.h" header only on UNIX (using __unix__) so that it will at least compile on Windows. The `nonTrivialAllocatorPopExternal` test will fail, however, it also does on master right now. 

A possible solution would be to move page size detection to the same place as cache line size detection and then provide it via pre-processor definition. As mentioned in my first PR, cache line detection does not currently work on Windows, however, I am planning to rectify this soon. While doing so, I could also add page size detection as a CMake configuration step.

Reply:
Perfect, will merge for now, I'll make a new patch to follow recommended way for page size, can't hurt. Then as you say, let's move this to the CMake config with your next patch.